dev-cpp/edencommon/edencommon-2025.04.14.00-r1.ebuild

```diff
@@ -22,13 +22,13 @@ SRC_URI="https://github.com/facebookexperimental/edencommon/archive/refs/tags/v$
 LICENSE="MIT"
 SLOT="0/${PV}"
 KEYWORDS="~amd64 ~arm64"
-IUSE="llvm-libunwind test"
+IUSE="io-uring llvm-libunwind test"
 RESTRICT="!test? ( test )"
 
 RDEPEND="
 	dev-cpp/glog:=
-	~dev-cpp/folly-${PV}:=
-	~dev-cpp/fb303-${PV}:=
+	~dev-cpp/folly-${PV}:=[io-uring(+)=]
```

Comment:
Don't we need io-uring deps on all packages as well here if the functionality is used & liburing gets linked against?

Reply:
~~Almost none of these packages are actually using io_uring. The problem is that to build them, a CMake file from Folly is required (`folly-targets.cmake` generated at build time), and this file includes `liburing.so` as a library to link against regardless of whether the dependent library actually uses it.~~

Actually, thinking about this again, we shouldn't need these USE dependencies I think, given that none of these packages (except fbthrift) depend on Folly's io_uring support. I'll drop the ones that are not required.

Reply:
Also, in the case of fbthrift we only ever use the Folly wrapper classes so we don't need a direct liburing dependency there either.

Reply:
(We also shouldn't rely on `-Wl,--as-needed` for correctness, so if linking against it, we should depend on it.)

Reply:
In theory there's no reason for any Thrift library to link against liburing, as it can just rely on it transitively through Folly (it doesn't use any liburing symbols on its own). Does what you say still apply here?

Reply:
Nah, just was a remark if the previous / struck-through part was the case.

Reply:
For the record, with the default stage3 amd64 tarball, VDB doesn't complain when we don't have it and libtree doesn't show a direct link against liburing on other Facebook packages. There's also `no-as-needed` in flag-o-matic so I think the default is always `as-needed`.

Reply:
The issue is that unlike I think Debian, we don't patch it in to the toolchain at all, so it's only in LDFLAGS in profiles and it's easy for someone to accidentally override it.